File: app/src/main/java/com/kieran/winnipegbusbackend/URLParameter.java
package com.kieran.winnipegbusbackend;

import java.util.List;

public class URLParameter {
    private static final String FORMAT = "%s=%s";
    private String key;
    private String value;

    public URLParameter(String key, String value) {
        this.key = key;
        this.value = value;
    }

    public URLParameter(String key, List<Integer> numbers) {
        this.key = key;
        value = "";

        for (int i = 0; i < numbers.size(); i++) {
            value += Integer.toString(numbers.get(i));
            if (i < numbers.size() - 1)
                value += ",";
        }
    }

    @Override
    public String toString() {
        return String.format(FORMAT, key, value);
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/URLParameter.kt
package com.kieran.winnipegbusbackend

class URLParameter {
    private var key: String? = null
    private var value: String? = null

    constructor(key: String, value: String) {
        this.key = key
        this.value = value
    }

    constructor(key: String, numbers: List<Int>) {
        this.key = key
        value = numbers.joinToString(",")
    }

    override fun toString(): String {
        return String.format(FORMAT, key, value)
    }

    companion object {
        private val FORMAT = "%s=%s"
    }
}
